### PR TITLE
handling OOB data generation and verification now pushed down to PAL …

### PR DESCRIPTION
**Pull request by paul-szczepanek-arm**
Vincent requested OOB handling to be pushed down below PAL layer

### CODE REVIEW

**Comment by paul-szczepanek-arm** [comment]
The short answer is: yes, the long answer is: nooooo.
This would have to be a compile time option since it would otherwise create needless redundancy and multiplying builds would just complicate everything - not sure it's worth it.

I see it as an incentive for the manufacturer to opt for using the cordio implementation

**Comment by pan-** [comment]
> This would have to be a compile time option since it would otherwise create needless redundancy and multiplying builds would just complicate everything - not sure it's worth it.

If its a compile time option then there is no reason to multiply builds; I must be missing something.

File: features/FEATURE_BLE/ble/pal/SecurityDb.h [hunk 1]
```diff
@@ -286,33 +286,6 @@ class SecurityDb {
...
-     */
-    virtual const public_key_coord_t& get_public_key_y() = 0;
-
-    /**
-     * Set local public key.
-     *
-     * @param[in] public_key_x new public key value of the x coordinate
-     * @param[in] public_key_y new public key value of the y coordinate
-     */
-    virtual void set_public_key(
```
[the remark this answers is not in the document]
Comment (paul-szczepanek-arm) [reply features/FEATURE_BLE/ble/pal/SecurityDb.h]: As far as I understand: if we are not doing the checking we don't need to store anything, other than the LTK. The public and private keys are only used during pairing. The stack can generate a new one for each new pairing or reuse the old one if it wants to. Upon reconnection it only needs us to give it the LTK we have stored.

[the remark this answers is not in the document]
Comment (pan-) [reply features/FEATURE_BLE/ble/pal/SecurityDb.h]: Fair enough; however we may need to add a function that sets the debug key (see 2.3.5.6.1 in BLUETOOTH SPECIFICATION Version 4.2 Vol 3, Part H).

[the remark this answers is not in the document]
Comment (paul-szczepanek-arm) [reply features/FEATURE_BLE/ble/pal/SecurityDb.h]: is this going to be available through the user API?

[the remark this answers is not in the document]
Comment (donatieng) [reply features/FEATURE_BLE/ble/pal/SecurityDb.h]: Mmh not sure we need debug APIs at this stage?

**Pull request merged**